src/site/topic20.rst

```diff
@@ -101,6 +101,30 @@ Iterative
 Recursive
 ---------
 
+* Below is a recursive implementation of a binary search
+* Notice that, other than being recursive, the underlying high-level algorithm is the same as the iterative implementation
+
+.. code-block:: java
+    :linenos:
+
+    public static <T extends Comparable<? super T>> int recursiveBinarySearch(T needle, T[] haystack, int lowIndex, int highIndex) {
+        if (lowIndex >= highIndex) {
+            return -1;
+        }
+        int midpoint = lowIndex + (highIndex - lowIndex) / 2;
+        if (haystack[midpoint].equals(needle)) {
+            return midpoint;
+        } else if (haystack[midpoint].compareTo(needle) > 0) {
+            highIndex = midpoint - 1;
+        } else {
+            lowIndex = midpoint + 1;
+        }
+        return recursiveBinarySearch(needle, haystack, lowIndex, highIndex);
```

Comment:
A little strange to modify the arguments. I'd rather see something like `return recursiveBinarySearch(needle, haystack, midpoint + 1, highIndex)` in the conditional blocks.

Reply:
I will make a new PR to change this in the class and in the content.